src/js/relayclient/ServerHelper.js

```diff
@@ -0,0 +1,168 @@
+     * { relayUrl: url,
+     *   RelayServerAddress: address
+     * }
+     */
+    async getRelayAddressPing(relayUrl) {
+        let self = this
+        return new Promise(function (resolve, reject) {
+            let callback = function (error, body) {
+                if (error) {
+                    reject(error);
+                    return
+                }
+                try {
+                    body.relayUrl = relayUrl
+                    resolve(body);
+                }
+                catch (err) {
+                    reject(err);
+                }
+            }
+            self.httpSend.send(relayUrl + "/getaddr", {}, callback)
+        });
+    }
+
+    /**
+     * From https://stackoverflow.com/a/37235207 (modified to catch exceptions)
+     * Resolves once any promise resolves, ignores the rest, ignores rejections
+     */
+    async raceToSuccess(promises) {
+        let numRejected = 0;
+        return new Promise(
+            (resolve, reject) =>
+                promises.forEach(
+                    promise =>
+                        promise.then((res) =>
+                            resolve(res)
+                        )
+                            .catch(err => {
+                                // console.log(err)
+                                if (++numRejected === promises.length) {
+                                    reject("No response from any server.", err);
+                                }
+                            })
+                )
+        );
+    }
+}
+
+class ServerHelper {
+    /**
+     * 
+     * @param {*} minStake 
+     * @param {*} minDelay 
+     * @param {*} httpSend 
+     */
+    constructor(minStake, minDelay, httpSend) {
+        this.minStake = minStake
+        this.minDelay = minDelay
+        this.httpSend = httpSend
+
+        this.filteredRelays = []
+        this.isInitialized = false
+    }
+    /**
+     * @param {*} relayHub 
+     * @param {*} relayHubAddress 
+     */
+    setHub(relayHub) {
+        //todo: re-run only on hub
+        this.relayHub = relayHub
+        addPastEvents(this.relayHub)
+        this.relayHubAddress = this.relayHub.address
+    }
+
+    async newActiveRelayPinger() {
+        if (this.relayHub === 'undefined') {
+            throw new Error("Must call to setHub first!")
+        }
+        if (this.filteredRelays.length == 0)
+        {
+            await this.fetchRelaysAdded()
+        }
+        return new ActiveRelayPinger(this.filteredRelays, this.httpSend)
+    }
+
+    /**
+     * Iterates through all RelayAdded and RelayRemoved logs emitted by given hub
+     * initializes an array {@link filteredRelays} of relays curently registered on given RelayHub contract
+     */
+    async fetchRelaysAdded() {
+        let activeRelays = {}
+        let addedAndRemovedEvents = await this.relayHub.getPastEvents({ address: this.relayHub.address, fromBlock: 1, topics: [["RelayAdded", "RelayRemoved"]] })
```

Comment:
don't scan from block 1. That's good for testing only.
estimate start block based on 12-second per block, e.g: `max(1,currentBlock - 24*60*(60/12))`